### PR TITLE
Linux string correctness

### PR DESCRIPTION
**Pull request by tbkka**
This works around a couple of bugs in the String type on Linux:
* `String.data()` crashes for empty strings -- workaround is to test for an empty string first
* `NSString(bytes:length:encoding:)` stops at zero bytes on Linux -- workaround is to explicitly look for a zero byte and use a much slower method when we see one.  (Note that even with the additional check for a zero byte, it's still much faster to use `NSString(bytes:length:encoding:)` when it does work.)

The first bug above seems to be fixed in newer snapshots, but I believe it still appears in the most recent official release build.

The second bug has been reported: https://bugs.swift.org/browse/SR-4216

### CODE REVIEW

**Comment by tbkka** [comment]
This fixes most (but not all) of the failures reported in #260.

File: Sources/SwiftProtobuf/StringUtils.swift [hunk 1]
```diff
@@ -17,7 +17,37 @@ import Foundation
 // Convert a pointer/count that refers to a block of UTF8 bytes
 // in memory into a String.
 // Returns nil if UTF8 is invalid.
+
+// This is painfully slow but seems to work correctly on every platform.
```
Comment (thomasvl) [inline Sources/SwiftProtobuf/StringUtils.swift]: Update comment since it is only used for linux?

File: Sources/SwiftProtobuf/StringUtils.swift [hunk 2]
```diff
@@ -17,7 +17,37 @@ import Foundation
...
+    }
+}
+#endif
+
 internal func utf8ToString(bytes: UnsafePointer<UInt8>, count: Int) -> String? {
+    if count == 0 {
+        return ""
+    }
+#if os(Linux)
+    // As of March, 2017, the NSString(bytes:length:encoding:)
```
Comment (thomasvl) [inline Sources/SwiftProtobuf/StringUtils.swift]: Bug link to track getting it fixed?

Comment (tbkka) [inline Sources/SwiftProtobuf/StringUtils.swift]: Done.

File: Sources/SwiftProtobuf/StringUtils.swift (hunk 1 above)
Comment (tbkka) [inline Sources/SwiftProtobuf/StringUtils.swift]: Done.

**Pull request merged**